src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java

```diff
@@ -0,0 +1,99 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.Serializable;
+
+/**
+ * A thread mode settings for the checkstyle modules.
+ * @author Andrew Kuchev
+ */
+public class ThreadModeSettings implements Serializable {
+    /** A checker module name. */
+    public static final String CHECKER_MODULE_NAME = Checker.class.getSimpleName();
+
+    /** A multi thread checker module name. */
+    public static final String MULTI_THREAD_CHECKER_MODULE_NAME =
+            Checker.class.getSimpleName();
+
+    /** A three walker module name. */
+    public static final String TREE_WALKER_MODULE_NAME = TreeWalker.class.getSimpleName();
+
+    /** A multi thread three walker module name. */
+    public static final String MULTI_THREAD_TREE_WALKER_MODULE_NAME =
+            TreeWalker.class.getSimpleName();
```

Comment:
@romani See my comment at https://github.com/checkstyle/checkstyle/pull/4420#issuecomment-306282276 and https://github.com/checkstyle/checkstyle/pull/4420#issuecomment-307835884.
Original code change removed all methods in `TreeWalker` and moved them to `AbstractTreeWalker` which is a huge and meaningless change as we have no implementation yet for `MultiThreadTreeWalker` and aren't sure what methods will be in it or which will be truly `abstract`.
I feel we should hold off on this until we have concrete design of multi-threading.